### PR TITLE
[Issue #59] Navigation component for Patient Profile

### PR DESCRIPTION
**Pull request by AlicjaKowalska**
Added navigation graph and partly implemented navigation for bottom nav bar.
Added HomeFragment.kt which replaces which implements the code of MainActivity.kt in order to maintain functionality of navigation bar.
Removed DrugActivity.kt as it is redundant when using navigation component.
Commented out code causing crash because of the changes to be implemented later on.

### CODE REVIEW

**Comment by piotrgos45** [comment]
When I try to look at drug details, I have screen like this:

image

**Comment by AlicjaKowalska** [comment]
> When I try to look at drug details, I have screen like this:
> 
> image

This should be covered in this PR [https://github.com/michalmobi/PatientAssistant/pull/77](url)

File: app/src/main/java/com/example/patientassistant/view/DrugActivity.kt [hunk 1]
```diff
@@ -17,56 +17,10 @@ import com.example.patientassistant.databinding.ActivityDrugBinding
 class DrugActivity : AppCompatActivity(R.layout.activity_drug) {
```
Comment (kozych) [inline app/src/main/java/com/example/patientassistant/view/DrugActivity.kt:17]: This file can be deleted as we do not use this class anymore

File: app/src/main/java/com/example/patientassistant/view/MainActivity.kt [hunk 2]
```diff
@@ -17,20 +17,29 @@ import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
```
Comment (kozych) [inline app/src/main/java/com/example/patientassistant/view/MainActivity.kt:17]: Please optimize imports

File: app/src/main/java/com/example/patientassistant/fragment/PatientProfileFragment.kt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        binding.floatingActionButtonAdd.setOnClickListener {
+            binding.floatingActionButtonAdd.visibility = View.INVISIBLE
+            binding.addInformation.visibility = View.VISIBLE
+            binding.background.visibility = View.VISIBLE
+        }
+
+        binding.floatingActionButtonCancel.setOnClickListener {
+            binding.addInformation.visibility = View.INVISIBLE
+            binding.floatingActionButtonAdd.visibility = View.VISIBLE
+        }
```
Comment (kozych) [inline app/src/main/java/com/example/patientassistant/fragment/PatientProfileFragment.kt:43]: Please add `binding.background.visibility = View.GONE`

File: app/src/main/java/com/example/patientassistant/view/MedicalAppointmentsActivity.kt [hunk 4]
```diff
@@ -19,73 +21,11 @@ import com.example.patientassistant.objects.MedicalAppointmentConstants
 class MedicalAppointmentsActivity() : AppCompatActivity() {
```
Comment (kozych) [inline app/src/main/java/com/example/patientassistant/view/MedicalAppointmentsActivity.kt:21]: This file can also be removed

File: app/src/main/res/layout/activity_drug.xml [hunk 5]
```diff
@@ -6,29 +6,5 @@
     android:layout_height="match_parent"
     tools:context=".view.DrugActivity">
```
Comment (kozych) [inline app/src/main/res/layout/activity_drug.xml:7]: please remove this file

File: app/src/main/res/layout/activity_medical_appointments.xml [hunk 6]
```diff
@@ -6,41 +6,5 @@
     android:layout_height="match_parent"
     tools:context=".view.MedicalAppointmentsActivity">
```
Comment (kozych) [inline app/src/main/res/layout/activity_medical_appointments.xml:7]: please remove this file

File: app/src/main/res/menu/menu_navigation.xml [hunk 7]
```diff
@@ -2,26 +2,26 @@
...
 
     <item
-        android:id="@+id/appointments"
+        android:id="@+id/appointmentListFragment"
         android:icon="@drawable/appointment"
         android:title="@string/appointments" />
 
     <item
-        android:id="@+id/map"
+        android:id="@+id/mapActivity3"
```
Comment (kozych) [inline app/src/main/res/menu/menu_navigation.xml:19]: Please change id to `mapActivity`

File: app/src/main/res/navigation/nav_graph.xml [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    <fragment
+        android:id="@+id/temperatureFragment"
+        android:name="com.example.patientassistant.fragment.TemperatureFragment"
+        android:label="fragment_temperature"
+        tools:layout="@layout/fragment_temperature" />
+    <fragment
+        android:id="@+id/weightFragment"
+        android:name="com.example.patientassistant.fragment.WeightFragment"
+        android:label="fragment_weight"
+        tools:layout="@layout/fragment_weight" />
```
Comment (kozych) [inline app/src/main/res/navigation/nav_graph.xml:78]: please remove numbers from ids

File: app/src/main/res/navigation/nav_graph.xml [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+            app:destination="@id/drugListFragment" />
+        <action
+            android:id="@+id/action_homeFragment_to_appointmentListFragment"
+            app:destination="@id/appointmentListFragment" />
+        <action
+            android:id="@+id/action_homeFragment_to_mapActivity3"
+            app:destination="@id/mapActivity3" />
+        <action
+            android:id="@+id/action_homeFragment_to_patientProfileFragment"
+            app:destination="@id/patientProfileFragment" />
```
Comment (kozych) [inline app/src/main/res/navigation/nav_graph.xml:24]: I think these are redundant, as these actions are never used

File: app/src/main/java/com/example/patientassistant/view/DrugActivity.kt (hunk 1 above)
Comment (AlicjaKowalska) [reply app/src/main/java/com/example/patientassistant/view/DrugActivity.kt]: Done, also commented the lines using DrugActivity in FragmentDrugForm.kt

File: app/src/main/java/com/example/patientassistant/view/MainActivity.kt (hunk 2 above)
Comment (AlicjaKowalska) [reply app/src/main/java/com/example/patientassistant/view/MainActivity.kt]: Done

File: app/src/main/java/com/example/patientassistant/fragment/PatientProfileFragment.kt (hunk 3 above)
Comment (AlicjaKowalska) [reply app/src/main/java/com/example/patientassistant/fragment/PatientProfileFragment.kt]: Done

File: app/src/main/res/layout/activity_drug.xml (hunk 5 above)
Comment (AlicjaKowalska) [reply app/src/main/res/layout/activity_drug.xml]: Done

File: app/src/main/res/layout/activity_medical_appointments.xml (hunk 6 above)
Comment (AlicjaKowalska) [reply app/src/main/res/layout/activity_medical_appointments.xml]: Done

File: app/src/main/res/menu/menu_navigation.xml (hunk 7 above)
Comment (AlicjaKowalska) [reply app/src/main/res/menu/menu_navigation.xml]: Done

File: app/src/main/res/navigation/nav_graph.xml (hunk 8 above)
Comment (AlicjaKowalska) [reply app/src/main/res/navigation/nav_graph.xml]: Done

File: app/src/main/res/navigation/nav_graph.xml (hunk 9 above)
Comment (AlicjaKowalska) [reply app/src/main/res/navigation/nav_graph.xml]: Removed redundant actions

**Review by kozych — APPROVED** [review]
Please resolve conflicts

**Review by piotrgos45 — APPROVED** [review]

File: app/src/main/java/com/example/patientassistant/workers/NotificationWorker.kt [hunk 10]
```diff
@@ -18,10 +18,10 @@ class NotificationWorker(context: Context, params: WorkerParameters) : Worker(co
         val isNotificationsEnabled = prefs.getBoolean("drugNotification", true)
 
         if (drugName != null && drugDose != null && drugId != null && isNotificationsEnabled) {
-            val drugNotificationHandler = DrugNotificationHandler(applicationContext)
+            /*val drugNotificationHandler = DrugNotificationHandler(applicationContext)
             drugNotificationHandler.createReminderNotification(
                 applicationContext, drugName, drugDose, drugId
-            )
+            )*/
```
Comment (michalmobi) [inline app/src/main/java/com/example/patientassistant/workers/NotificationWorker.kt:24]: why is this commented out?

**Comment by michalmobi** [comment]
In this PR there are many changes that do not strictly relate to ticket #59. For example "HomeFragment" is added and I as a reviewer don't know why. Please add a short explanation here or in the ticket why so many changes were needed.

File: app/src/main/java/com/example/patientassistant/workers/NotificationWorker.kt (hunk 10 above)
Comment (AlicjaKowalska) [reply app/src/main/java/com/example/patientassistant/workers/NotificationWorker.kt:24]: It caused crash because of removal of the DrugActivity.kt

**Comment by AlicjaKowalska** [comment]
> In this PR there are many changes that do not strictly relate to ticket #59. For example "HomeFragment" is added and I as a reviewer don't know why. Please add a short explanation here or in the ticket why so many changes were needed.

Added explanation in the PR comment above

**Pull request merged**